frontstage/views/account/account_survey_share.py

```diff
@@ -44,38 +42,42 @@ def share_survey_business_select(session):
 @account_bp.route('/share-surveys/business-selection', methods=['POST'])
 @jwt_authorization(request)
 def share_survey_post_business_select(session):
-    form = AccountSurveyShareBusinessSelectForm(request.values)
-    if not form.validate():
-        flash('Choose an answer')
+    share_survey_business_selected = request.form.getlist("checkbox-answer")
+    if len(share_survey_business_selected) == 0:
+        flash('You need to choose a business')
         return redirect(url_for('account_bp.share_survey_business_select'))
-    flask_session['share_survey_business_selected'] = form.data['option']
+    flask_session['share'] = {k: [] for k in share_survey_business_selected}
     return redirect(url_for('account_bp.share_survey_survey_select'))
 
 
 @account_bp.route('/share-surveys/survey-selection', methods=['GET'])
 @jwt_authorization(request)
 def share_survey_survey_select(session):
     party_id = session.get_party_id()
-    surveys = get_surveys_listed_against_party_and_business_id(flask_session['share_survey_business_selected'],
-                                                               party_id)
+    share_dict = {}
+    for key in flask_session['share']:
+        selected_business = get_business_by_id(key)
+        surveys = get_surveys_listed_against_party_and_business_id(key,
+                                                                   party_id)
+        share_dict[selected_business[0]['name']] = surveys
     is_max_share_survey = request.args.get('max_share_survey', False)
-    flask_session['share_survey_surveys_selected'] = None
-    selected_business = get_business_by_id([flask_session['share_survey_business_selected']])
+    failed_survey_key = request.args.get('key', None)
     return render_template('surveys/surveys-share/survey-select.html',
-                           surveys=surveys, business_name=selected_business[0]['name'],
-                           is_max_share_survey=is_max_share_survey)
+                           share_dict=share_dict, is_max_share_survey=is_max_share_survey,
+                           failed_survey_key=failed_survey_key)
 
 
-def validate_max_shared_survey():
+def validate_max_shared_survey(key: str, share_survey_surveys_selected: list):
```

Comment:
can `key` just be `business_id` if that's what it is?  It makes it easier to read

Reply:
👍 +1